zipline/data/us_equity_minutes.py

```diff
@@ -11,27 +11,401 @@
+            compression ratios are not ideal.
+
+            Defaults to supporting 15 years of NYSE equity market data.
+
+            see: http://bcolz.blosc.org/opt-tips.html#informing-about-the-length-of-your-carrays
+        """
+        self._rootdir = rootdir
+        self._first_trading_day = first_trading_day
+        self._market_opens = market_opens[
+            market_opens.index.slice_indexer(start=self._first_trading_day)]
+        self._trading_days = market_opens.index
+        self._minutes_per_day = minutes_per_day
+        self._expectedlen = expectedlen
+        self._ohlc_ratio = ohlc_ratio
+
+        self._minute_index = _calc_minute_index(
+            self._market_opens, self._minutes_per_day)
+
+        self._write_metadata()
+
+    @property
+    def first_trading_day(self):
+        return self._first_trading_day
+
+    @property
+    def metadata_path(self):
+        return os.path.join(self._rootdir, METADATA_FILENAME)
+
+    def _write_metadata(self):
+        """
+        Write the metadata to a JSON file in the rootdir.
+
+        Values contained in the metadata are:
+        first_trading_day : string
+            'YYYY-MM-DD' formatted representation of the first trading day
+             available in the dataset.
+        minute_index : list of integers
+             nanosecond integer representation of the minutes, the enumeration of which
+             corresponds to the values in each bcolz carray.
+        """
+        metadata = {
+            'first_trading_day': str(self.first_trading_day.date()),
+            'minute_index': self._minute_index.asi8.tolist(),
+            'ohlc_ratio': self._ohlc_ratio,
+        }
+        with open(self.metadata_path, 'w+') as fp:
+            json.dump(metadata, fp)
+
+    def sidpath(self, sid):
+        """
+        Parameters:
+        -----------
+        sid : int
+            Asset identifier.
+
+        Returns:
+        --------
+        out : string
+            Full path to the bcolz rootdir for the given sid.
+        """
+        sid_subdir = _sid_subdir_path(sid)
+        return join(self._rootdir, sid_subdir)
+
+    def last_date_in_output_for_sid(self, sid):
+        """
+        Parameters:
+        -----------
+        sid : int
+            Asset identifier.
+
+        Returns:
+        --------
+        out : pd.Timestamp
+            The midnight of the last date written in to the output for the
+            given sid.
+        """
+        try:
+            sizes_path = "{0}/close/meta/sizes".format(self.sidpath(sid))
+            with open(sizes_path, mode='r') as f:
+                sizes = f.read()
+            data = json.loads(sizes)
+            num_days = data['shape'][0] / self._minutes_per_day
+            if num_days == 0:
+                # empty container
+                return None
+            return self._trading_days[num_days - 1]
+        except IOError:
```

Comment:
This is hit in the case when the ctable has not yet been created.
However, before calling this we can just check for ctable existence.